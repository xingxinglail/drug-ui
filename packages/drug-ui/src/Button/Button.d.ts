import * as React from 'react';
export declare type Variant = 'text' | 'outlined' | 'contained' | 'fab';
export declare type Color = 'default' | 'primary' | 'secondary' | 'inherit';
export declare type Size = 'small' | 'medium' | 'large';
export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
    variant?: Variant;
    color?: Color;
    size?: Size;
    disabled?: boolean;
    disableRipple?: boolean;
    fullWidth?: boolean;
    href?: string;
    round?: boolean;
    fab?: boolean;
    icon?: boolean;
    loading?: boolean;
    children?: React.ReactNode;
    component?: React.ElementType;
    ref?: React.Ref<HTMLButtonElement>;
}
declare const Button: React.FC<ButtonProps>;
export default Button;
