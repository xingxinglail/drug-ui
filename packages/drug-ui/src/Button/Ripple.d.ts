import * as React from 'react';
interface RippleProps {
    center?: boolean;
}
declare const Ripple: React.FC<RippleProps>;
export default Ripple;
