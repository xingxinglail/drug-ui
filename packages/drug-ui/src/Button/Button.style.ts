import { Style, Styles } from 'jss';
import { Theme } from '../styles';

const root: Style = {
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative',
    color: 'rgba(0, 0, 0, 0.87)',
    padding: [6, 16],
    fontSize: 14,
    minWidth: 64,
    boxSizing: 'border-box',
    transition: 'background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
    fontFamily: `"Roboto", "Helvetica", "Arial", sans-serif`,
    fontWeight: 500,
    lineHeight: 1.75,
    borderRadius: 4,
    letterSpacing: 0.4,
    verticalAlign: 'middle',
    border: 0,
    outline: 0,
    cursor: 'pointer',
    textDecoration: 'none',
    userSelect: 'none',
    '-webkit-appearance': 'none',
    '-webkit-tap-highlight-color': 'transparent',
    '&:hover': {
        textDecoration: 'none',
        backgroundColor: 'rgba(0, 0, 0, 0.08)'
    },
    '&$disabled': {
        cursor: 'default',
        pointerEvents: 'none',
        color: 'rgba(0, 0, 0, 0.26)'
    }
};

const contained: Style = () => ({
    boxShadow: '0 3px 1px -2px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12)',
    backgroundColor: '#e0e0e0',
    '&:hover': {
        boxShadow: '0 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12)',
        backgroundColor: '#d5d5d5'
    },
    '&$disabled': {
        boxShadow: 'none',
        backgroundColor: 'rgba(0, 0, 0, 0.12)'
    }
});

const containedPrimary: Style = (theme: Theme) => ({
    color: theme.palette.primary.contrastText,
    backgroundColor: theme.palette.primary.main,
    '&:hover': {
        backgroundColor: '#115293'
    }
});

const containedSecondary: Style = (theme: Theme) => ({
    color: theme.palette.secondary.contrastText,
    backgroundColor: theme.palette.secondary.main,
    '&:hover': {
        backgroundColor: '#9a0036'
    }
});

const text: Style = {
    padding: [6, 8]
};

const textPrimary: Style = {
    color: '#1976d2',
    '&:hover': {
        backgroundColor: 'rgba(25, 118, 210, 0.08)'
    }
};

const textSecondary: Style = {
    color: '#dc004e',
    '&:hover': {
        backgroundColor: 'rgba(220, 0, 78, 0.08)'
    }
};

const outlined: Style = {
    border: '1px solid rgba(0, 0, 0, 0.23)',
    padding: [5, 15],
    '&:hover': {
        backgroundColor: 'rgba(0, 0, 0, 0.08)'
    },
    '&$disabled': {
        border: '1px solid rgba(0, 0, 0, 0.26)'
    }
};

const outlinedPrimary: Style = {
    color: '#1976d2',
    border: '1px solid rgba(25, 118, 210, 0.5)',
    '&:hover': {
        border: '1px solid 1976d2',
        backgroundColor: 'rgba(25, 118, 210, 0.08)'
    }
};

const outlinedSecondary: Style = {
    color: '#dc004e',
    border: '1px solid rgba(220, 0, 78, 0.5)',
    '&:hover': {
        border: '1px solid rgb(220, 0, 78)',
        backgroundColor: 'rgba(220, 0, 78, 0.08)'
    }
};

const colorInherit: Style = {
    color: 'inherit',
    borderColor: 'currentColor'
};

const fab: Style = {
    minWidth: 'unset',
    width: 48,
    height: 48,
    borderRadius: '50%',
    padding: 0,
    boxShadow: '0 3px 1px -2px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12)',
    backgroundColor: '#e0e0e0',
    '&:hover': {
        boxShadow: '0 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12)',
        backgroundColor: '#d5d5d5'
    },
    '&$disabled': {
        boxShadow: 'none',
        backgroundColor: 'rgba(0, 0, 0, 0.12)'
    }
};

const fullWidth: Style = {
    width: '100%'
};

const disabled: Style = {};

const icon: Style = {
    ...fab,
    boxShadow: 'none',
    backgroundColor: null,
    '&:hover': null
};

export const styles = (theme: Theme): Styles => ({
    root: root,
    contained: contained(),
    containedPrimary: containedPrimary(theme),
    containedSecondary: containedSecondary(theme),
    text: text,
    textPrimary: textPrimary,
    textSecondary: textSecondary,
    outlined: outlined,
    outlinedPrimary: outlinedPrimary,
    outlinedSecondary: outlinedSecondary,
    colorInherit: colorInherit,
    fab: fab,
    fabPrimary: containedPrimary(theme),
    fabSecondary: containedSecondary(theme),
    icon,
    fullWidth,
    disabled
});
