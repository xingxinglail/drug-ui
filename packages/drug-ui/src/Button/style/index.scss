.drug-button {
    $primary: #1976d2;
    $secondary: #dc004e;
    $disabled: rgba(0, 0, 0, 0.26);
    $small-font-size: 13px;
    $large-font-size: 15px;

    display: inline-flex;
    align-items: center;
    justify-content: center;
    position: relative;
    color: rgba(0, 0, 0, 0.87);
    padding: 6px 16px;
    font-size: 14px;
    min-width: 64px;
    box-sizing: border-box;
    transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms, border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
    font-weight: 500;
    line-height: 1.75;
    border-radius: 4px;
    letter-spacing: 0.4px;
    vertical-align: middle;
    border: 0;
    outline: 0;
    cursor: pointer;
    text-decoration: none;
    user-select: none;
    -webkit-appearance: none;
    -webkit-tap-highlight-color: transparent;

    .drug-icon + .drug-button-label {
        margin-left: 8px;
    }

    .drug-button-label + .drug-icon {
        margin-left: 8px;
    }

    &-round {
        border-radius: 24px;
    }

    &:hover {
        text-decoration: none;
        background-color: rgba(0, 0, 0, 0.08);
    }

    &-full-width {
        width: 100%;
    }

    &-contained, &-fab {
        box-shadow: 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12);
        background-color: #e0e0e0;

        &:hover {
            box-shadow: 0 2px 4px -1px rgba(0, 0, 0, 0.2), 0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12);
            background-color: #d5d5d5;
        }

        &-primary {
            color: #fff;
            background-color: $primary;;

            &:hover {
                background-color: #115293;
            }
        }

        &-secondary {
            color: #fff;
            background-color: $secondary;

            &:hover {
                background-color: #9a0036;
            }
        }

        &-small {
            padding: 4px 10px;
            font-size: $small-font-size;
        }

        &-large {
            padding: 8px 22px;
            font-size: $large-font-size;
        }

        &-disabled {
            color: $disabled;
            box-shadow: none;
            background-color: rgba(0, 0, 0, 0.12);
        }
    }

    &-text {
        padding: 6px 8px;

        &-primary {
            color: $primary;

            &:hover {
                background-color: rgba(25, 118, 210, 0.08);
            }
        }

        &-secondary {
            color: $secondary;

            &:hover {
                background-color: rgba(220, 0, 78, 0.08);
            }
        }

        &-small {
            padding: 4px 5px;
            font-size: $small-font-size;
        }

        &-large {
            padding: 8px 11px;
            font-size: $large-font-size;
        }

        &-disabled {
            color: $disabled;
        }
    }

    &-outlined {
        border: 1px solid rgba(0, 0, 0, 0.23);
        padding: 5px 15px;

        &:hover {
            background-color: rgba(0, 0, 0, 0.08);
        }

        &-primary {
            color: $primary;
            border: 1px solid rgba(25, 118, 210, 0.5);

            &:hover {
                border: 1px solid $primary;
                background-color: rgba(25, 118, 210, 0.08);
            }
        }

        &-secondary {
            color: $secondary;
            border: 1px solid rgba(220, 0, 78, 0.5);

            &:hover {
                border: 1px solid rgb(220, 0, 78);
                background-color: rgba(220, 0, 78, 0.08);
            }
        }

        &-small {
            padding: 3px 9px;
            font-size: $small-font-size;
        }

        &-large {
            padding: 7px 21px;
            font-size: $large-font-size;
        }

        &-disabled {
            color: $disabled;
            border: 1px solid $disabled;
        }
    }

    &-fab, &-icon {
        min-width: unset;
        width: 48px;
        height: 48px;
        border-radius: 50%;
        padding: 0;

        &-small {
            width: 40px;
            height: 40px;
        }

        &-large {
            width: 56px;
            height: 56px;
        }
    }

    &-color-inherit {
        color: inherit;
        border-color: currentColor;
    }

    &-loading {

        @keyframes spin {

            100% {
                -webkit-transform: rotate(360deg);
                transform: rotate(360deg)
            }
        }

        .icon-spin {
            animation: spin 1s infinite linear;
        }

        &::before {
            content: '';
            position: absolute;
            top: -1px;
            right: -1px;
            bottom: -1px;
            left: -1px;
            z-index: 1;
            background: #fff;
            border-radius: inherit;
            opacity: .35;
            -webkit-transition: opacity .2s;
            transition: opacity .2s;
        }
    }

    &-disabled, &-loading {
        cursor: default;
        pointer-events: none;
    }
}
