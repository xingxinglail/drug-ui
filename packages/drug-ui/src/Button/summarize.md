### Button 总结

1. React.forwardRef
    - 要获取子组件的 ```ref``` 必须使用 ```React.forwardRef``` 来创建一个组件
  
2. React.Children
    - ```React.Children``` 可以遍历 ```props.children``` 可以对每一个 children 进行额外处理。 

3. 类型
    - ```FC``` 函数组件类型。
    - ```ReactNode``` 所有 children 类型。
    - ```ElementType``` html 标签名或者函数组件。
    - ```HTMLAttributes``` html 标签属性。
