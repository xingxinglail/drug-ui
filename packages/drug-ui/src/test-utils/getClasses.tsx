import * as React from 'react';
import * as enzyme from 'enzyme';
import Wrapper from './Wrapper';

export const getClasses = (element: React.ReactElement) => {
    // @ts-ignore
    const { useStyles } = element.type;
    let classes;

    function Listener () {
        classes = useStyles();
        return null;
    }

    enzyme.mount(<Wrapper><Listener /></Wrapper>);
    return classes;
};
