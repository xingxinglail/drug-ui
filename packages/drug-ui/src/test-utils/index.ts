export { default as Wrapper } from './Wrapper';
export { getClasses } from './getClasses';

export const wait = (delay = 50) => {
    return new Promise(resolve => setTimeout(resolve, delay));
};
