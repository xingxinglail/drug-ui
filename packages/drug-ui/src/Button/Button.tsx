import * as React from 'react';
import * as PropTypes from 'prop-types';
import { PropTypes as ComponentPropTypes } from '..';
import classnames from 'classnames';
import { createUseStyles } from '@drug-ui/styles';
import { Loading } from '@drug-ui/icons';
import { capitalize } from '../utils';
import { styles } from './Button.style';
import { Theme } from '../styles';
import ButtonBase, { ButtonBaseProps } from '../ButtonBase';

export type Variant = 'text' | 'outlined' | 'contained';

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement>, ButtonBaseProps {
    variant?: Variant;
    color?: ComponentPropTypes.Color;
    size?: ComponentPropTypes.Size;
    fullWidth?: boolean;
    round?: boolean;
    loading?: boolean;
    ref?: React.Ref<HTMLButtonElement>
}

type ButtonClassProps =
    'root'
    | 'contained'
    | 'containedPrimary'
    | 'containedSecondary'
    | 'text'
    | 'textPrimary'
    | 'textSecondary'
    | 'outlined'
    | 'outlinedPrimary'
    | 'outlinedSecondary'
    | 'fab'
    | 'fabPrimary'
    | 'fabSecondary'
    | 'icon'
    | 'colorInherit'
    | 'fullWidth'
    | 'disabled'
    | 'containedSizeSmall'
    | 'containedSizeLarge'
    | 'textSizeSmall'
    | 'textSizeLarge'
    | 'outlinedSizeSmall'
    | 'outlinedSizeLarge'
    | 'round'
    | 'label'
    | 'loading';

const name = 'Button';

const useStyles = createUseStyles<Theme, ButtonClassProps>(styles, { name });

const Button: React.FC<ButtonProps> = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
    const { className, variant = 'text', color = 'default', size = 'medium', disabled, component, fullWidth: defaultFullWidth, href, round, loading, children, ...rest } = props;
    const classes = useStyles();

    const classNames = classnames(
        classes.root,
        [classes[variant]],
        {
            [classes[`${ variant }${ capitalize(color) }`]]: color !== 'inherit' && color !== 'default',
            [classes[`${ variant }Size${ capitalize(size) }`]]: size !== 'medium',
            [classes.colorInherit]: color === 'inherit',
            [classes.fullWidth]: defaultFullWidth,
            [classes.round]: round,
            [classes.loading]: loading,
            [classes.disabled]: disabled
        }
    );

    const newChildren = React.Children.map(children, child => {
        const type = typeof child;
        if (type === 'string' || type === 'number') {
            return <span className={ classes.label }>{ child }</span>;
        }
        return child;
    });

    return (
        <ButtonBase
            className={ classNames }
            component={ component }
            ref={ ref }
            href={ href }
            disabled={ disabled }
            { ...rest }>
            <Loading/>
            { newChildren }
            {/*{ loading ? <Icons name="loading" className="loading"/> : null }*/ }
        </ButtonBase>
    );
});

Button.displayName = name;

Button.defaultProps = {
    variant: 'text',
    color: 'default',
    size: 'medium',
    fullWidth: false,
    loading: false,
    round: false
};

Button.propTypes = {
    variant: PropTypes.oneOf<Variant>(['text', 'outlined', 'contained']),
    color: PropTypes.oneOf<ComponentPropTypes.Color>(['default', 'primary', 'secondary', 'inherit']),
    size: PropTypes.oneOf<ComponentPropTypes.Size>(['small', 'medium', 'large']),
    loading: PropTypes.bool,
    fullWidth: PropTypes.bool,
    round: PropTypes.bool
};

export default Button;
