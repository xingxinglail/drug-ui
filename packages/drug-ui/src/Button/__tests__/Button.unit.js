import * as renderer from 'react-test-renderer';
import * as React from 'react';
import * as enzyme from 'enzyme';
import Button from '..';
import { Wrapper } from '@drug-ui/core/test-utils';

const mount = enzyme.mount;

describe('<Button />', () => {

    it('render successful', () => {
        const json = renderer.create(<Wrapper><Button /></Wrapper>).toJSON();
        expect(json).toMatchSnapshot();
    });

    it('可以设置 className', () => {
        const wrapper = mount(<Wrapper><Button className="button button2">Hello World</Button></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('button')).toBeTruthy();
        expect(classList.contains('button2')).toBeTruthy();
    });

    it('默认按钮', () => {
        const wrapper = mount(<Wrapper><Button><span>1</span></Button></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.length).toBe(3);
    });

    it('设置 variant = contained', () => {
        const wrapper = mount(<Wrapper><Button variant="contained" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-contained')).toBeTruthy();
        expect(classList.length).toBe(3);
    });

    it('设置 variant = outlined', () => {
        const wrapper = mount(<Wrapper><Button variant="outlined" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-outlined')).toBeTruthy();
        expect(classList.length).toBe(3);
    });

    it('设置 variant = xxx', () => {
        const wrapper = mount(<Wrapper><Button variant="xxx" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeFalsy();
        expect(classList.contains('DuiButton-contained')).toBeFalsy();
        expect(classList.contains('DuiButton-outlined')).toBeFalsy();
        expect(classList.length).toBe(2);
    });

    it('设置 color = inherit', () => {
        const wrapper = mount(<Wrapper><Button color="inherit" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-colorInherit')).toBeTruthy();
        expect(classList.length).toBe(4);
    });

    it('设置 color = primary', () => {
        const wrapper = mount(<Wrapper><Button color="primary" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-textPrimary')).toBeTruthy();
        expect(classList.length).toBe(4);
    });

    it('设置 color = secondary', () => {
        const wrapper = mount(<Wrapper><Button color="secondary" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-textSecondary')).toBeTruthy();
        expect(classList.length).toBe(4);
    });

    it('设置 color = xxx', () => {
        const wrapper = mount(<Wrapper><Button color="xxx" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        console.log(classList);
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-colorInherit')).toBeFalsy();
        expect(classList.contains('DuiButton-textPrimary')).toBeFalsy();
        expect(classList.contains('DuiButton-textSecondary')).toBeFalsy();
        expect(classList.length).toBe(3);
    });

    it('设置 size = small', () => {
        const wrapper = mount(<Wrapper><Button size="small" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-textSizeSmall')).toBeTruthy();
        expect(classList.length).toBe(4);
    });

    it('设置 size = large', () => {
        const wrapper = mount(<Wrapper><Button size="large" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-textSizeLarge')).toBeTruthy();
        expect(classList.length).toBe(4);
    });

    it('设置 size = xxx', () => {
        const wrapper = mount(<Wrapper><Button size="xxx" /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-textSizeSmall')).toBeFalsy();
        expect(classList.contains('DuiButton-textSizeLarge')).toBeFalsy();
        expect(classList.length).toBe(3);
    });

    it('设置 fullWidth', () => {
        const wrapper = mount(<Wrapper><Button fullWidth /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-fullWidth')).toBeTruthy();
        expect(classList.length).toBe(4);
    });

    it('设置 round', () => {
        const wrapper = mount(<Wrapper><Button round /></Wrapper>);
        const classList = wrapper.getDOMNode().classList;
        expect(classList.contains('DuiButtonBase-root')).toBeTruthy();
        expect(classList.contains('DuiButton-root')).toBeTruthy();
        expect(classList.contains('DuiButton-text')).toBeTruthy();
        expect(classList.contains('DuiButton-round')).toBeTruthy();
        expect(classList.length).toBe(4);
    });
});
