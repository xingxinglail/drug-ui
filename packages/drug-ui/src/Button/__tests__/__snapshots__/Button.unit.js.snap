// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render successful 1`] = `
<button
  className="DuiButtonBase-root DuiButton-root DuiButton-text"
  disabled={false}
>
  <span
    className="DuiRipple-root"
    onDragLeave={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
  />
</button>
`;
